app/styles/ui/_side-by-side-diff.scss

```diff
@@ -62,267 +55,315 @@
       }
-    }
-  }
 
-  .hunk-handle {
-    position: absolute;
-    height: 100%;
-    width: 20px;
-    left: calc(50% - 10px);
-    top: 0;
+      .selectable {
+        &.hoverable:hover {
+          border-color: var(--diff-add-hover-border-color);
+        }
 
-    &.hoverable {
-      cursor: pointer;
+        &.hoverable:hover,
+        &.hoverable:hover label,
+        &.hover,
+        &.hover label,
+        input:focus-visible + label {
+          background: var(--diff-add-hover-background-color);
+          color: var(--diff-add-hover-text-color);
+          border-color: var(--diff-add-hover-border-color);
+        }
+      }
     }
-  }
 
-  .content {
-    display: flex;
-    white-space: pre-wrap;
-    overflow-y: auto;
-    flex-grow: 1;
-    word-break: break-all;
-
-    .prefix {
-      user-select: none;
+    .hunk-handle-place-holder,
+    .hunk-handle {
+      width: var(--hunk-handle-width);
+      height: 100%;
+      background-color: var(--diff-empty-hunk-handle);
     }
 
-    .octicon {
-      height: 8px;
-      fill: var(--error-color);
-      display: inline-block;
-      margin-left: 3px;
-      margin-bottom: -1px;
+    &.has-check-all-control {
+      .hunk-handle-place-holder,
+      .hunk-handle {
+        width: var(--hunk-handle-width-with-check-all);
+      }
     }
-  }
 
-  .before,
-  .after {
-    width: 50%;
-    display: flex;
-    flex-direction: row;
-    position: relative;
-  }
+    .hunk-handle {
+      z-index: 1;
 
-  .before {
-    color: var(--diff-delete-text-color);
-    background: var(--diff-delete-background-color);
+      &.hoverable {
+        cursor: pointer;
+      }
 
-    .line-number {
-      background: var(--diff-delete-gutter-background-color);
-      border-color: var(--diff-delete-border-color);
+      .increased-hover-surface {
+        position: absolute;
+        margin-left: -5px;
+        width: 15px;
+        z-index: 1;
+        cursor: pointer;
+      }
     }
 
-    .selectable {
-      &.hoverable:hover {
-        border-color: var(--diff-delete-hover-border-color);
+    &.modified {
+      .increased-hover-surface {
+        width: 20px;
+        margin-left: -8px;
       }
+    }
 
-      &.hoverable:hover,
-      &.hoverable:hover label,
-      &.hover,
-      &.hover label,
-      input:focus-visible + label {
-        background: var(--diff-delete-hover-background-color);
-        color: var(--diff-delete-hover-text-color);
-        border-color: var(--diff-delete-hover-border-color);
+    &:has(.line-number.selectable.line-selected) {
+      .hunk-handle {
+        background-color: var(--diff-selected-border-color);
       }
     }
-  }
-
-  .after {
-    color: var(--diff-add-text-color);
-    background: var(--diff-add-background-color);
 
-    .line-number {
-      background: var(--diff-add-gutter-background-color);
-      border-color: var(--diff-add-border-color);
+    &:has(.line-number.selectable) {
```

Comment:
These are not quite as up to par functionality wise since it means the hunk handle selected state really only responds to the first line of the selection, I believe I will have to do some js for the check-all selection state for them to behave 100% as expected.  Next PR.